src/script/lua_api/l_inventory.cpp

```diff
@@ -273,18 +270,17 @@ int InvRef::l_set_lists(lua_State *L)
 	}
 
 	// Make a temporary inventory in case reading fails
-	Inventory *tempInv(inv);
-	tempInv->clear();
+	Inventory tempInv(inv->getItemDef());
 
 	Server *server = getServer(L);
 
 	lua_pushnil(L);
 	while (lua_next(L, 2)) {
 		const char *listname = lua_tostring(L, -2);
-		read_inventory_list(L, -1, tempInv, listname, server);
+		read_inventory_list(L, -1, &tempInv, listname, server);
 		lua_pop(L, 1);
 	}
-	inv = tempInv;
+	*inv = tempInv;
```

Comment:
I am hinting toward the fact that = might not do what you think it does.  It would probably be prudent to double check that all of the side effects resulting from this operator are intended.

Take a look at m_dirty.